h2o-automl/src/main/java/ai/h2o/automl/IAlgo.java

```diff
@@ -0,0 +1,9 @@
+package ai.h2o.automl;
+
+import java.io.Serializable;
+
+public interface IAlgo extends Serializable {
```

Comment:
is there a way to avoid using Serializable? it will work but serializable is meant more of a fallback

Reply:
for interfaces, this is the only possibility together with `Freezable`.
The problem is that implementing all Freezable methods each time we basically want to define a new string is a bit overkill...
Also, I'm not familiar enough with the Weaver internals, but the main implementation of this interface is the `Algo` enum, so I expect the Weaver to use the `enum` machinism to serialize the value and not the `Serializable` one: it all depends if given a class property, the Weaver uses the declared property type or the actual property type, this I don't know.

Reply:
the only reason this integface seems to exist is testing, that can be acomplished with a mock library or simply not using enum for Algo, which I think is worth considering since this IInterface naming scheme is very ugly and quite confusing
also having an interfac where we never expect another implementation in prod is not a great idea

Reply:
@honzasterba https://github.com/h2oai/h2o-3/pull/4375#discussion_r398239638
Note that `ModelingStep` supports external definitions, and in that regard, any algo name, even those not listed in the enum should be supported...
I'm considering getting rid of the enum though, it's even documented in the Algo definition.
For info, agree that the interface naming is not pretty, but at least it's consistent (cf. `IStoppingMetric` for another enum...).